pbench/server/hierarchy.py

```diff
@@ -0,0 +1,349 @@
+import os
+import time
+
+
+def filepermission(fileperm):
+    stat = ""
+    while fileperm > 0:
+        mod = fileperm % 10
+        permdict = {
+            0: "---",
+            1: "--x",
+            2: "-w-",
+            3: "-wx",
+            4: "r--",
+            5: "r-x",
+            6: "rw-",
+            7: "rwx",
+        }
+        res = permdict[mod]
+        stat = res + stat
+        fileperm = fileperm // 10
+
+    return f"-{stat}"
```

Comment:
"stat" is a file mode here, but "stat" is a start/stop indicator in the "header()" routine below. You should use different names.